components/pedigree/ui/src/main/resources/PhenoTips/UIX_Field__pedigree.xml

```diff
@@ -124,7 +124,13 @@
 {{velocity}}
 #if ($xcontext.action != 'export')
   #set ($family = $services.families.getFamilyForPatient($doc.name))
-  #pedigreeThumbnail($family, $doc.name)
+  #set ($familyRef = "$!doc.getObject('PhenoTips.FamilyReferenceClass').getValue('reference')")
+  #if ("$!family" == '' &amp;&amp; $familyRef != '' &amp;&amp; !$xwiki.getDocument($familyRef).isNew())
```

Comment:
just curious why do we use `isNew()` here?

Reply:
`$services.families.getFamilyForPatient` returns `null` in 2 cases:
- there is no family for that patient
- there is a family but the current user doesn't have access to it

So we need to figure out when the user doesn't have access to the family. That would be when the patient has a family reference, and that corresponds to an actual family document. OTOH, if the patient has some invalid family reference, then we should allow to create a new family.

Invalid family reference means the document doesn't exist (`$xwiki.getDocument($familyRef).isNew()`) or is not a family document. The document referred by family reference can be absent for example on XAR import, when you import a patient with a family reference but you don't import that family, or (I think) when you delete the family document using XWiki's delete functionality rather than using the families service.

(Now that I'm detailing this, we could also capture the situation where the reference document exists but it's not a family document, i.e. it doesn't have a family object, which can happen when manually deleting the family object from it. But that's far less likely to happen than the other 2 scenarios.)